src/ui/server/index.ts

```diff
@@ -144,23 +147,35 @@ const errorHandler: express.ErrorRequestHandler = (err, req, res, next) => {
 
 app.use(errorHandler);
 
-app.get('*', (req: Request, res: Response) =>
-  res.render('page-not-found.njk', {
-    CONTENT_STRINGS: {
-      COOKIES_CONSENT,
-      FOOTER: QUOTE_FOOTER,
-      LINKS,
-      PHASE_BANNER,
-      PRODUCT: {
-        ...PRODUCT,
-        DESCRIPTION: PRODUCT.DESCRIPTION.GENERIC,
-      },
-      ...PAGES.PAGE_NOT_FOUND_PAGE,
-    },
-    START_ROUTE: ROUTES.QUOTE.START,
-    FEEDBACK_ROUTE: LINKS.EXTERNAL.FEEDBACK,
-  }),
-);
+const INSURANCE_PAGE_NOT_FOUND_TEMPLATE = 'insurance/page-not-found.njk';
+const PAGE_NOT_FOUND_TEMPLATE = 'page-not-found.njk';
+
+app.get('*', (req: Request, res: Response) => {
+  /**
+   * if route contains "insurance" eg /insurance/dashboard
+   * insurance page-not-found should be rendered
+   */
+  if (isInsuranceRoute(req.originalUrl)) {
+    return res.render(INSURANCE_PAGE_NOT_FOUND_TEMPLATE, {
+      ...corePageVariables({
+        PAGE_CONTENT_STRINGS: PAGES.PAGE_NOT_FOUND_PAGE,
+        ORIGINAL_URL: req.originalUrl,
+      }),
+      userName: getUserNameFromSession(req.session.user),
+    });
+  }
+
+  // all other non-insurance page not found should use generic page-not-found template
```

Comment:
How does it work for quote pages/routes?

Reply:
Should we not have something like `isQuoteRoute(...)`? which then has quote links in the header and footer?

Reply:
done